### ISSUE TITLE
Unhandled Runtime Error - "Error: Cancel rendering route"

### ISSUE THREAD

**Issue by funk10101** [supertokens/supertokens-core#536]
## 🐛 Bug Report
This is an intermittent error. In my site, I click on the SignIn/SignUp button:
- The ST SignIn form shows. I click on the "Sign Up" link and the Sign Up form shows fine.
- If I click on the Sign Up form again, I get this "Unhandled Runtime Error - "Error: Cancel rendering route" error.
Please check my code, and see the screen shots

My github repo (with just the files in question):
https://github.com/funk10101/nextjs-supertokens-issue

## Useful informations
- NextJS 13.0.5
- supertokens-auth-react ^0.27.3
- supertokens-node ^12.1.1

![SignInScreen](https://user-images.githubusercontent.com/119277851/204209994-f3fb300e-1e37-4639-a1dc-b84cc53890ee.jpg)
![SignUpScreen](https://user-images.githubusercontent.com/119277851/204210040-364670d9-1a18-4413-b2c1-176bb3dff025.jpg)
![SignUpScreen-Error](https://user-images.githubusercontent.com/119277851/204210059-48c77799-99ea-4a0d-98ff-ca8a112467d8.jpg)

**Comment by funk10101** [comment]
Update: I commented out ` windowHandler: (oI) => {
      return {
        ...oI,
        location: {
          ...oI.location,
          setHref: (href) => {
            Router.push(href);
          },
        },
      };
    },`

and so far it hasn't given me the error.

**Comment by porcellus** [comment]
TLDR: I recommend using an `a` tag instead of `Link` to redirect to auth.
As a side note, you might want to avoid calling `redirectToAuth` with `redirectBack: true` on the auth page because it'll overwrite where the user is supposed to end up after logging in.

More details:
After experimenting with this for a bit, it looks like the error means that two redirections are going on at the same time.
I could reproduce this only when:
- I'm using the Link tag from "next/link"
- The href of the link tag is not empty (it's set to "#" in your example)
- `Router.push` is called in the `onClick` of the Link tag. (which `redirectToAuth` would do with the `setHref` override)

So there are a few options to resolve this:
1. Use an `a` tag instead.
2. Set the `href` to an empty string (although this also seems to produce some weird behaviour).
3. Remove the `setHref` override (which means you get full page redirects, so it's not the best)

All in all, I'd recommend you avoid this issue by using an `a` tag instead of the `Link`.

**Comment by rishabhpoddar** [comment]
@porcellus is there something we can do in our auth-react SDK to resolve this? As opposed to constraining the user to use certain ways of doing things?

**Comment by rishabhpoddar** [comment]
Right ok. So i'm closing this issue now since no action is required from our side.
